packages/Tag/src/Tag.js

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Close from "@paprika/icon/lib/Times";
+import * as sc from "./Tag.styles";
+import * as types from "./types";
+
+export function Delete({ isDisabled, size, onRemove = () => {} }) {
+  return (
+    <sc.Delete isDisabled={isDisabled} size={size} data-pka-anchor="tag.delete" onClick={onRemove}>
+      <Close />
+    </sc.Delete>
+  );
+}
+
+Delete.propTypes = {
+  onRemove: PropTypes.func.isRequired,
+  size: PropTypes.oneOf([types.MEDIUM, types.LARGE]).isRequired,
+  isDisabled: PropTypes.bool,
+};
+
+Delete.defaultProps = {
+  isDisabled: false,
+};
+
+export default function Tag(props) {
+  const { children, onRemove, size, onClick, a11yText, isDisabled, ...moreProps } = props;
+  const isChildString = typeof children === "string";
+
+  const handleRemove = e => {
+    onRemove();
+    e.stopPropagation();
+  };
+
+  const content = (
+    <>
+      <sc.Ellipsis isChildString={isChildString} size={size}>
+        {children}
+      </sc.Ellipsis>
+      {onRemove ? <Delete isDisabled={isDisabled} onRemove={handleRemove} size={size} /> : null}
```

Comment:
We had discussed about this several time before, sadly I don't remember what was the agreements, cc @mikrotron do you remember or do we have something about this in our wiki? I remember @jamiek-acl pushed for using the function as an indicator to enable button.

If we follow the input API you should include the prop hasDeleteButton as hasClearButton. And just for that reason, my vote would go to consistency.

Reply:
I think it was decided to _NOT_ do that (though I think it is silly).
I think the example was the old Multiselect, when passing in `onFilter` it would show the filter field (no need for a `showFilter` prop, too).

Reply:
I remember we made the decision to have an explicit `has` prop during the days of `acl-ui`, but then changed our minds for Paprika. I think it was Jamie's argument that convinced enough of us during ShopTalk... but it was long enough ago that I don't remember the details, and it doesn't seem to be captured in the wiki.  But I think it was in relation to the API for the `<DialogActions>` component, and it currently follows the same pattern as Tristan has here, so I think we should stick with it:

https://github.com/acl-services/paprika/blob/6b3f25e0cc77cfa9fe2829bf07781bab703431e8/packages/DialogActions/src/DialogActions.js#L51